### PR TITLE
wip - convert lunch-configuration to lunch-template

### PR DESCRIPTION
**Pull request by zhou322**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -149,6 +175,31 @@ resource "aws_autoscaling_group" "this_with_initial_lifecycle_hook" {
   protect_from_scale_in     = var.protect_from_scale_in
   service_linked_role_arn   = var.service_linked_role_arn
 
+  mixed_instances_policy {
```
Comment (martijnblankestijn) [inline main.tf:238]: why the duplication of the 2 autoscaling groups ("this" and "this_with_initial_lifecycle_hook")?

File: main.tf [hunk 2]
```diff
@@ -96,6 +97,31 @@ resource "aws_autoscaling_group" "this" {
...
+          weighted_capacity = lookup(override.value, "weighted_capacity", null)
+        }
+      }
+    }
+
+    instances_distribution {
+      on_demand_base_capacity                  = var.on_demand_base_capacity
+      on_demand_percentage_above_base_capacity = var.on_demand_percentage_above_base_capacity
+      spot_allocation_strategy                 = var.spot_allocation_strategy
+      spot_instance_pools                      = var.spot_allocation_strategy == "lowest-price" ? var.spot_instance_pools : null
```
Comment (martijnblankestijn) [inline main.tf:178]: doesn't the instance pools also have to be defined in the capacity-optimized allocation strategy?

File: variables.tf [hunk 3]
```diff
@@ -321,3 +321,44 @@ variable "max_instance_lifetime" {
   default     = 0
 }
 
+variable "instance_types" {
```
Comment (martijnblankestijn) [inline variables.tf:324]: As these are new variables, where are they set (or is this part WIP)?

File: main.tf (hunk 2 above)
Comment (zhou322) [reply main.tf:178]: no. it is only for `lowest-price`  strategy

File: variables.tf (hunk 3 above)
Comment (zhou322) [reply variables.tf:324]: they are set in the maistro-infra repo.

File: main.tf (hunk 1 above)
Comment (zhou322) [reply main.tf:238]: u can check [
https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/38](url) for more detail.

**Pull request merged**